org.alloytools.alloy.dash/src/main/java/ca/uwaterloo/watform/rapidDash/DashExprToPython.java

```diff
@@ -0,0 +1,310 @@
+                break;
+            case SOME_ARROW_LONE:
+                res = " ";
+                break;
+            case ONE_ARROW_ANY:
+                res = " ";
+                break;
+            case ONE_ARROW_SOME:
+                res = " ";
+                break;
+            case ONE_ARROW_ONE:
+                res = " ";
+                break;
+            case ONE_ARROW_LONE:
+                res = " ";
+                break;
+            case LONE_ARROW_ANY:
+                res = " ";
+                break;
+            case LONE_ARROW_SOME:
+                res = " ";
+                break;
+            case LONE_ARROW_ONE:
+                res = " ";
+                break;
+            case LONE_ARROW_LONE:
+                res = " ";
+                break;
+            case ISSEQ_ARROW_LONE:
+                res = " ";
+                break;
+            case JOIN:
+                res = " ";
+                break;
+            case DOMAIN:
+                res = " ";
+                break;
+            case RANGE:
+                res = " ";
+                break;
+            case INTERSECT:
+                res = " ";
+                break;
+            case PLUSPLUS:
+                res = " ";
+                break;
+            case PLUS:
+                res = " + ";
+                break;
+            case IPLUS:
+                res = " ";
+                break;
+            case MINUS:
+                res = " - ";
+                break;
+            case IMINUS:
+                res = " ";
+                break;
+            case MUL:
+                res = " ";
+                break;
+            case DIV:
+                res = " ";
+                break;
+            case REM:
+                res = " ";
+                break;
+            case EQUALS:
+                res = " == ";
+                break;
+            case NOT_EQUALS:    // TODO: or "is not"?
+                res = " != ";
+                break;
+            case IMPLIES:
+                res = " ";
+                break;
+            case LT:
+                res = " < ";
+                break;
+            case LTE:
+                res = " <= ";
+                break;
+            case GT:
+                res = " > ";
+                break;
+            case GTE:
+                res = " >= ";
+                break;
+            case NOT_LT:
+                res = " >= ";
+                break;
+            case NOT_LTE:
+                res = " > ";
+                break;
+            case NOT_GT:
+                res = " <= ";
+                break;
+            case NOT_GTE:
+                res = " < ";
+                break;
+            case SHL:
+                res = " ";
+                break;
+            case SHA:
+                res = " ";
+                break;
+            case SHR:
+                res = " ";
+                break;
+            case IN:
+                res = " in ";
```

Comment:
Alloy operator "in" means subset, while Python operator "in" checks if a value is present in a sequence (list, range, string etc.). I don't think "in" can be directly translated into "in", remember everything in Alloy is "set".

Reply:
Then what should it be translated into then?

Reply:
We want to check if a set is in a set, then we should use `in` to check that, isn't it?

Reply:
E.g., if we have signatures `SigA in SigB` and we define them to be `SigA: set = set(); SigB: set = set()`, which we should do since they are sets in Dash. Then we can do `SigA in SigB` to check `IN` operation in python

Reply:
Can in operator in Python be used like that? like {1,2} in {1,2,3}. I don’t think in can be used to check subset relationship.

Reply:
I am thinking about maybe check the type of left and right argument, then depending on the argument type, do a case switch statement…

Reply:
no it doesn't.

Reply:
As I mentioned later, that's the work of translating the signatures, this is only the translation of the operation symbol. It needs the translation of signatures to come up with an executable code. After all, this is the operation between two signature instances that we created.

Reply:
See my comments below.

Reply:
Assume everything is a set violates the manual translation of `Elevator.py`, but sure, if you think that's the direction, I am fine

Reply:
Yeah, that’s why I propose checking the types of left right arguments, see whether they are objects, sets or integers. And translate into proper operators.